### PR TITLE
.gitignore and configuration housekeeping

### PR DESCRIPTION
**Pull request by dewiniaid**
This performs some minor housekeeping:
- Use a more robust .gitignore
- Use sopel.cfg-dist as the sample configuration file.  Removes sopel.cfg from the repository, so developers need not worry about losing their working copies.
- Fleshes out the config file with optional parameters and comments and suggestions on how to set it up properly on a new checkout.
- Adds a common 'ratbot' configuration section for defining settings that should be shared to all rat-\* modules, like API URLs and such.
- Allows specification of a 'workdir' folder (default 'run') for data that is modified at runtime, and update modules that modify files (namely !search) to use workdir when determining paths for those files.

This is against current sopel-master, so lacks any of the changes in other PRs -- they'll need to be updated against this.

### CODE REVIEW

File: sopel.cfg-dist [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## Admins
+# owner = OwnerNameHere
+# admins = Joe,Bob
+
+## Where to find modules
+extra = sopel-modules
+
+## For production/testing
+enable = admin,help,rat-board,rat-facts,rat-search
```
Comment (tyrope) [inline sopel.cfg-dist:24]: Reload should be in production. It's limited to bot.admin/bot.owner only. A reload is faster than an entire bot restart and doesn't interfere with other modules.

File: sopel.cfg-dist [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+db_filename = run/ratbot.db
+logdir = logs
+verbose = verbose
+log_raw = True
+
+[ratbot]
+## Common configuration to all rat modules.
+# If running sopel in a read-only location, make sure it has write access to workdir
+workdir = run
+apiurl = http://dev.api.fuelrats.com
```
Comment (tyrope) [inline sopel.cfg-dist:38]: This should (like above) seperate Development/Testing (dev.api.) and Production (api.)

File: sopel.cfg-dist [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+[ratboard]
```
Comment (tyrope) [inline sopel.cfg-dist]: Why the empty section?

**Comment by dewiniaid** [comment]
I was having a bad day where, in the midst of this, I kept losing my sopel.cfg file randomly.  I may have gotten a bit sloppy on my 5th time recreating it.  The empty section is where the api URL was before I moved it to be something that's not exclusive to the rescue board.

Reload is a security thing -- since owner/admins can only be validated by nick and I don't consider that secure.  

As for dev vs prod API URLs: That's something that can be remedied.  Dev should be the default, though I really hope whoever ultimately ends up hosting this doesn't just blindly copy the config file without looking at it first.

**Comment by dewiniaid** [comment]
Updated based on feedback.  Should be good to merge now.

**Pull request closed without merging**